link.go

```diff
@@ -28,6 +28,14 @@ type LinkAttrs struct {
 	Namespace    interface{} // nil | NsPid | NsFd
 }
 
+// NewLinkAttrs returns LinkAttrs structure filled with default values
+func NewLinkAttrs() LinkAttrs {
+	return LinkAttrs{
+		TxQLen: -1,
+		MTU:    -1,
+	}
+}
+
```

Comment:
If we can't actually set zero for either then we could just leave zero as
the default, but I thought the initial bug was that it doesn't default to
10000

Vish

On Fri, Jun 12, 2015, 2:52 PM Alexander Morozov <notifications@github.com>
wrote:

> In link.go
> <https://github.com/vishvananda/netlink/pull/35#discussion_r32358495>:
>
> > @@ -28,6 +28,14 @@ type LinkAttrs struct {
> >  	Namespace    interface{} // nil | NsPid | NsFd
> >  }
> >
> > +// NewLinkAttrs returns LinkAttrs structure filled with default values
> > +func NewLinkAttrs() LinkAttrs {
> > +	return LinkAttrs{
> > +		TxQLen: -1,
> > +		MTU:    -1,
> > +	}
> > +}
> > +
>
> Hmm, I'm not sure, but seems like we can't set TxQLen to 0 with netlink.
> iproute2 uses ioctl.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vishvananda/netlink/pull/35/files#r32358495>.
>